src/main/scala/tech/cryptonomic/conseil/Harpoon.scala

```diff
@@ -0,0 +1,142 @@
+package tech.cryptonomic.conseil
+
+import cats.implicits._
+import cats.effect.{ExitCode, IO, IOApp}
+import tech.cryptonomic.conseil.process.AccountsTaggingProcess
+import tech.cryptonomic.conseil.process.AccountsTaggingProcess.BlockLevel
+import tech.cryptonomic.conseil.tezos.{Tables, TezosDatabaseOperations, TezosTypes}
+import tech.cryptonomic.conseil.tezos.TezosDatabaseOperations.AccountFlags
+import tech.cryptonomic.conseil.util.{DatabaseUtil, PureLogging}
+import tech.cryptonomic.conseil.config.HarpoonAppConfig
+import slick.jdbc.PostgresProfile.api._
+import scala.concurrent.ExecutionContext
+import tech.cryptonomic.conseil.util.IOUtil
+import slick.jdbc.ResultSetType
+import slick.jdbc.ResultSetConcurrency
+import cats.data.StateT
+import cats.mtl.implicits._
+import cats.~>
+
+/** Runs an independet process that scans existing operations
+  * recorded from the chainm to look for reveal or activation
+  * and thus mark the corresponding Account record
+  */
+object Harpoon extends IOApp with PureLogging {
+
+  /** A type alias to provide state management on top of IO.
+    * The state threaded through corresponds to the actual
+    * process input value (AccountsTaggingProcess.ProcessorInput i.e. Option[BlockLevel]),
+    * which allows to start processing from that point onward
+    */
+  type IOState[T] = StateT[IO, Option[BlockLevel], T]
+
+  /* reference to the database to use, we need write access to update accounts */
+  private lazy val dbHandle: Database = DatabaseUtil.lorreDb
+
+  /* This is the main entrypoint, the return type is wrapped in IO and provides an ExitCode for the process
+   * Described in cats-effect apis: https://typelevel.org/cats-effect/api/cats/effect/IOApp.html
+   */
+  override def run(args: List[String]): IO[ExitCode] = {
+
+    //we read custom conf, usually provided from the environment
+    val conf = HarpoonAppConfig
+
+    implicit val ctx = scala.concurrent.ExecutionContext.global
+    val timer = IO.timer(scala.concurrent.ExecutionContext.global)
+
+    //we create the processor stage, with the custom types needed
+    val tagging = AccountsTaggingProcess[IOState]
+
+    /* Implement any dependency needed to run the processing. */
+    implicit val serviceApis = injectServices(tagging)
+
+    /* now we can call the processing stage */
+    val processCycle = (processedTopLevel: tagging.ProcessorInput) =>
+      tagging.process
+        .run(processedTopLevel)
+        .flatTap {
+          case (level, flagged) =>
+            logger.pureLog[IO](
```

Comment:
Might want to use log4cats for nicer syntax

Reply:
Interesting, does it support lightbend loggers?
I've gone with a home-made very straightforward wrapper, I'll look into that

Reply:
It doesn't support them, afaik. You'd use string interpolation/concatenation to insert parameters.

Reply:
Thanks, I'll keep it in mind, but not for this PR.
I want to keep the current logging library here, will need a more through investigation of which logging to use on a dedicated task.
There we might consider switching to log4cats on top of the logging of choice.
😉